heft-plugins/heft-jest-plugin/src/JestConfigLoader.ts

```diff
@@ -0,0 +1,239 @@
+import * as path from 'path';
+import merge from 'deepmerge';
+import { Import, JsonFile } from '@rushstack/node-core-library';
+
+import type { Config } from '@jest/types';
+
+/**
+ *
+ */
+export class JestConfigLoader {
+  private static readonly _rootDirRegex: RegExp = /<rootDir>/g;
+  private static readonly _defaultReporter: string = 'DEFAULT';
+
+  /**
+   * Load the full text of the final Jest config file, substituting specified tokens for
+   * provided values.
+   */
+  public static async loadConfigAsync(filePath: string, rootDir: string): Promise<Config.InitialOptions> {
+    let config: Config.InitialOptions = await JestConfigLoader._readConfigAsync(filePath);
+
+    // If a preset exists, let's load it manually and unset the preset string so that Jest normalization
+    // is not affected. This means that we can support a couple features here that stock Jest doesn't have:
+    // - support loading non-root presets
+    // - support for loading presets further than 1 level deep
+    if (config.preset) {
+      const presetConfigPath: string = JestConfigLoader._resolveConfigModule(
+        config.preset,
+        filePath,
+        rootDir,
+        'preset'
+      );
+      const presetConfig: Config.InitialOptions = await JestConfigLoader._loadPresetAndResolveModulesAsync(
+        presetConfigPath,
+        rootDir
+      );
+      config = JestConfigLoader._mergeConfig(config, presetConfig);
+      config.preset = undefined;
+    }
+
+    return config;
+  }
+
+  private static async _readConfigAsync(configPath: string): Promise<Config.InitialOptions> {
+    try {
+      return await JsonFile.loadAsync(configPath);
+    } catch (e) {
+      throw new Error(`Could not load Jest config at path "${configPath}".`);
```

Comment:
I'd have separate error messages for not-found and other errors. In the latter case, we should print the error.